docs/validation.md

```diff
@@ -0,0 +1,76 @@
+# Validation & Assertions
+
+## Inputs
+
+Within the confines of a 'step', adding the inputs parameter allows the user to specify the same name within the `cmd` tag as a standard `bash` variable (`$VARNAME`), 
+which creates the possibility to chain inputs from one step to the next.
+
+The format is simply to specify a name.
+
+```
+- name: Cat added file
+    on_node: 1
+    inputs:
+      - HASH
+    cmd: ipfs cat $HASH
+```
+
+## Outputs
+
+With in the confines of a 'step', adding the 'outputs' parameter allows the user to send the output from the command on a given line to a given name.
+This can then be verified later to validate the output of a particular command.
+
+The format is for a given `line`, save the output to the named variable `save_to`.
```

Comment:
You should try to find a way to incorporate the `append_to` feature in this discussion too, perhaps referencing the iteration section.